bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/internal/CommunicationManagerTest.java

```diff
@@ -176,13 +188,22 @@ public void setup() {
         manager.addProfileFactory(mockProfileFactory);
         manager.addProfileAdvisor(mockProfileAdvisor);
 
-        ItemChannelLinkRegistry iclRegistry = new ItemChannelLinkRegistry() {
+        ItemChannelLinkRegistryAdvanced iclRegistry = new ItemChannelLinkRegistryAdvanced();
+        iclRegistry.addProvider(new ItemChannelLinkProvider() {
+            @Override
+            public void addProviderChangeListener(@NonNull ProviderChangeListener<@NonNull ItemChannelLink> listener) {
+            }
+
             @Override
-            public Stream<ItemChannelLink> stream() {
-                return Arrays.asList(LINK_1_S1, LINK_1_S2, LINK_2_S2, LINK_1_T1, LINK_1_T2, LINK_2_T2, LINK_3_S3)
-                        .stream();
+            public void removeProviderChangeListener(
+                    @NonNull ProviderChangeListener<@NonNull ItemChannelLink> listener) {
             }
-        };
+
+            @Override
+            public @NonNull Collection<@NonNull ItemChannelLink> getAll() {
```

Comment:
Eclipse always likes adding `@NonNull`, but it shouldn't be used according to the [conventions](https://www.eclipse.org/smarthome/documentation/development/conventions.html).

Reply:
You are right. It has been added by the Eclipse IDE code generation and I missed to remove it again.